### PR TITLE
Static analysis of play-java subproject.

### PR DESCRIPTION
**Pull request by KoenDG**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [x] Have you added tests for any changed functionality?

Don't think this actually changed functionality.

# Helpful things

## Purpose

This is not related to any issue.

I dive into the source code every now and then to look up how things are done and I recently found a small thing where I thought "that's not the ideal way to use a for loop". Seeing as how reading the source code had been helpful for me, I thought I'd do some static analysis, find some stuff to clean up and then submit said cleanup. Kinda like the boy scout rule, which is mentioned in the 'contributing' page.

This PR limits itself to the `play-java` sub-project.

## References

No references

So yeah. general cleanup. It makes the code a bit cleaner. I fully understand if you're all like "we don't actually need this right now".

Things that changed:
* \* imports changed to fully qualified imports.
* Seemingly pointless `matcher.matches();` calls removed.
* 2 duplicate primitive assignments fixed
* for loop into while loop, which seems more logical for the intended functionality of that bit of code
* Use of JDK8's `computeIfAbsent` functionality, which is expressly intended for "check if map index is null, if so add something at that index" scenarios.
* Removing redundant escapes in a regex. That's according to Intellij.

### CODE REVIEW

File: framework/src/play-java/src/main/java/play/libs/Time.java [hunk 1]
```diff
@@ -41,19 +41,15 @@ public static int parseDuration(String duration) {
         int toAdd = -1;
         if (days.matcher(duration).matches()) {
             Matcher matcher = days.matcher(duration);
-            matcher.matches();
```
Comment (gmethvin) [inline framework/src/play-java/src/main/java/play/libs/Time.java]: My understanding of the `Matcher` API is that you need to call `matches()` in order for `group` to work.

Whatever the case is, if you're going to change this code I would like to see a test verifying that the behavior stays the same.

Comment (wsargent) [inline framework/src/play-java/src/main/java/play/libs/Time.java]: Yeah, the matcher is stateful. https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#matches()

**Comment by KoenDG** [comment]
PR updated. I returned the `matches` calls. Wasn't aware of that functionality. Added unit tests for the Time test. Copyright section at the top copy-pasted from another test.

Also undid the removing of the redundant escapes in the pattern matching regex in RoutingDsl.java
It doesn't have a unit test and I don't see how I could add one for that class. So leaving that as-is.

File: framework/src/play-java/src/main/java/play/libs/XPath.java [hunk 2]
```diff
@@ -58,13 +67,9 @@ public Iterator getPrefixes(String namespaceURI) {
...
             final String p = requireNonNull(prefix, "Null prefix");
             final String uri = requireNonNull(namespaceURI, "Null namespaceURI");
-            if (! XMLConstants.DEFAULT_NS_PREFIX.equals(p)) {
+            if (!XMLConstants.DEFAULT_NS_PREFIX.equals(p)) {
                 prefixMap.put(p, uri);
-                Set<String> prefixSet = namespaceMap.get(uri);
-                if (prefixSet == null) {
-                    prefixSet = new LinkedHashSet<>();
-                    this.namespaceMap.put(uri, prefixSet);
-                }
```
Comment (KoenDG) [inline framework/src/play-java/src/main/java/play/libs/XPath.java]: `computeIfAbsent` was created to do exactly this functionality. Far as I'm aware, the `XpathSpec.scala` file acts as unit test for this functionality.

File: framework/src/play-java/src/main/java/play/routing/RoutingDsl.java [hunk 3]
```diff
@@ -331,7 +335,7 @@ public boolean tryAdvance(Consumer<? super MatchResult> action) {
     }
 
     private static final Pattern paramExtractor =
-            Pattern.compile("([:\\*\\$])(\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*)(?:<(.*)>)?");
+            Pattern.compile("([:*$])(\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*)(?:<(.*)>)?");
```
Comment (KoenDG) [inline framework/src/play-java/src/main/java/play/routing/RoutingDsl.java]: Redundant character escape.

Seeing as `AbstractRoutingDslTest`/`DependencyInjectedRoutingDslTest` does a bunch of tests with RoutingDsl methods that depend on this, that's tested, far as I can tell.

**Pull request merged**